server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -2808,6 +2808,9 @@ public void doInTransactionWithoutResult(final TransactionStatus status) {
                             Pair<String, Long> storeUrlAndId = new Pair<>(url, store.getId());
                             for (HypervisorType hypervisorType : hypSet) {
                                 try {
+                                    if (HypervisorType.Simulator == hypervisorType) {
+                                        continue;
+                                    }
```

Comment:
I don't like that simulator is referenced in production code. Not sure what the solution would be in this case. Open for suggestions.